src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/ExtMethods.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+namespace Microsoft.Rest.ClientRuntime.Azure.TestFramework
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+
+    public static class ExtMethods
+    {
+        /// <summary>
+        /// Allow to get value for key (either title case or lowercase)
+        /// This allows users to set the connection string without worrying about case sensitivity of the keys in the key-value pairs within
+        /// connection string
+        /// </summary>
+        /// <param name="dictionary"></param>
+        /// <param name="keyName"></param>
+        /// <returns></returns>
+        public static string GetValueUsingCaseInsensitiveKey(this Dictionary<string, string> dictionary, string keyName)
+        {
+            string valueForKey;
+            if (dictionary.TryGetValue(keyName, out valueForKey))
+            {
+                return valueForKey;
+            }
+            else if (dictionary.TryGetValue(keyName.ToLower(), out valueForKey))
+            {
+                return valueForKey;
+            }
+
+            return valueForKey;
+        }
+
+        /// <summary>
+        /// Searches dictionary with key as provided as well as key.ToLower()
+        /// </summary>
+        /// <param name="dictionary"></param>
+        /// <param name="keyName"></param>
+        /// <returns></returns>
+        public static bool ContainsCaseInsensitiveKey(this Dictionary<string, string> dictionary, string keyName)
+        {
+            if (dictionary.ContainsKey(keyName))
```

Comment:
as stated above.

---
In reply to: [78655171](https://github.com/Azure/azure-sdk-for-net/pull/2328#discussion_r78655171) [](ancestors = 78655171)